app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php

```diff
@@ -725,4 +725,109 @@ function () use ($emailMock) {
         $errorMessages = $model->getErrors();
         $this->assertCount(1, $errorMessages);
     }
+
+    public function testSetSampleMailerByTemporaryMailer()
+    {
+        // Use our test token transport limiting to 1 recipient per queue
+        $transport = new BatchTransport(true, 1);
+        $mailer    = new \Swift_Mailer($transport);
+
+        // Mock factory to ensure that queue mode is handled until MailHelper is refactored completely away from MauticFactory
+        $factoryMock = $this->getMockBuilder(MauticFactory::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $factoryMock->method('getParameter')
+            ->willReturnCallback(
+                function ($param) {
+                    switch ($param) {
```

Comment:
It's better to avoid any logic in the test. To avoid it in this case, use:

```
$factoryMock->expects($this->at(0))->method('getParameter')->with('mailer_spool_type')->willReturn('memory');
$factoryMock->expects($this->at(1))->method('getParameter')->willReturn(''));
```
This code might not work right away. The X `$this->at(X)` means the call index of the class, not the method. It makes it a bit trickier to get the X right.